### PR TITLE
Panel

### PR DESCRIPTION
**Pull request by dsaliba**
Im sorry

PS. I dont know why it still pushes all theese bin files

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Panel.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.core751.wrappers.WCANSparkMax;
+import frc.robot.core751.wrappers.WVictorSPX;
+
+public class Panel extends SubsystemBase {
+
+    public enum WheelColor {
+        BLUE(240, 255, 255, true),
```
Comment (InsightGit) [inline src/main/java/frc/robot/subsystems/Panel.java:23]: I was wondering when you would use a enum in your PR

File: src/main/java/frc/robot/subsystems/Panel.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                case 'G' :
+                return GREEN;
+                case 'R' :
+                return RED;
+                case 'Y' :
+                return YELLOW;
+                default :
+                return UNKNOWN;
+            }
+            } else {
```
Comment (InsightGit) [inline src/main/java/frc/robot/subsystems/Panel.java:50]: could you also MAYBE format your code?

File: src/main/java/frc/robot/subsystems/Panel.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+       
+       return WheelColor.UNKNOWN;
+    }
+
+    public WheelColor getFirstColor() {
+        Color firstDetectedColor = firstColorSensor.getColor();
+
+        ColorMatchResult firstMatch = colorMatcher.matchClosestColor(firstDetectedColor);
+        
+
```
Comment (InsightGit) [inline src/main/java/frc/robot/subsystems/Panel.java]: um why the whitespace

File: src/main/java/frc/robot/subsystems/Panel.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    public double[] getAverageColors() {
+        Color firstDetectedColor = firstColorSensor.getColor();
+        Color secondDetectedColor = secondColorSensor.getColor();
+        double[] ret = new double[]{firstDetectedColor.red + secondDetectedColor.red, firstDetectedColor.green + secondDetectedColor.green, firstDetectedColor.blue + secondDetectedColor.blue};
+        ret[0] /=2;
+        ret[1] /=2;
+        ret[2] /=2;
+        return ret;
+
```
Comment (InsightGit) [inline src/main/java/frc/robot/subsystems/Panel.java]: refer to the above comment about whitespace

File: src/main/java/frc/robot/commands/Panel/RotateWheel.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        this.lightStrip.update();
+        if (this.wheelColor == WheelColor.BETWEEN) {
+            this.panel.setSpinMotor(1);
+        }
+        // if (this.wheelColor == WheelColor.UNKNOWN) {
+        //     this.panel.stopSpinMotor();
+        //     this.finished = true;
+        // }
+        
+        double speed = Math.abs(Math.abs(this.target) - Math.abs(this.panel.getRotations()))<=0.125f?0.5:1;
```
Comment (InsightGit) [inline src/main/java/frc/robot/commands/Panel/RotateWheel.java]: thank you dimitri very cool ? operator. Though could you maybe not for readability?

File: src/main/java/frc/robot/commands/Panel/GoToColor.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            this.lightStrip.setHSVWave(i, 2, c);
+        }
+        this.lightStrip.update();
+        this.lightStrip.start();
+        this.panel.clearRotations();
+        this.targetColor = WheelColor.getTargetColor();
+    }
+
+    @Override
+
```
Comment (InsightGit) [inline src/main/java/frc/robot/commands/Panel/GoToColor.java:43]: please cease and desist with the whitespace

**Review by InsightGit — CHANGES_REQUESTED** [review]
Looks ok, however, please format your code and maybe don't use ? operators kthx

File: src/main/java/frc/robot/subsystems/Panel.java (hunk 1 above)
Comment (InsightGit) [reply src/main/java/frc/robot/subsystems/Panel.java]: java enums give me will to live

Comment (dsaliba) [reply src/main/java/frc/robot/subsystems/Panel.java:23]: Java enums give me will to live

File: src/main/java/frc/robot/subsystems/Panel.java (hunk 2 above)
Comment (dsaliba) [reply src/main/java/frc/robot/subsystems/Panel.java:50]: in my defense that was a direct example from WPILIb

File: src/main/java/frc/robot/subsystems/Panel.java (hunk 3 above)
Comment (dsaliba) [reply src/main/java/frc/robot/subsystems/Panel.java]: Fixed white space other than some left in for readability

File: src/main/java/frc/robot/commands/Panel/RotateWheel.java (hunk 5 above)
Comment (dsaliba) [reply src/main/java/frc/robot/commands/Panel/RotateWheel.java]: Fixed it 😞

**Comment by dsaliba** [comment]
Addressed all code style changes

**Review by InsightGit — APPROVED** [review]
All good. I'll merge it in!

**Pull request merged**